### PR TITLE
Tests for Numba 0.54.0 RC3

### PR DESCRIPTION
**Pull request by guilhermeleobas**
Changes:
* Update RBC code to work with Numba 0.54
* Upgrade tests to use the `omnisci` fixture
* RemoteJIT actions run after OmniSciDB ones
* OmniSciDB tests only run OmniSciDB specific tests. The same is true for RemoteJIT
* RemoteJIT actions stops if any container fails
* One can use a star to specify all possible values of an intrinsic. i.e. `experimental.vector.reduce.fmin.*` is reduced to [all possible values](https://github.com/llvm-mirror/llvm/blob/master/test/Analysis/CostModel/AArch64/vector-reduce.ll)

### CODE REVIEW

File: rbc/externals/stdio.py [hunk 1]
```diff
@@ -13,7 +14,10 @@
 def cg_fflush(builder):
     int8_t = ir.IntType(8)
     fflush_fnty = ir.FunctionType(int32_t, [int8_t.as_pointer()])
-    fflush_fn = builder.module.get_or_insert_function(fflush_fnty, name="fflush")
+    if get_version('numba') >= (0, 54):
+        fflush_fn = cgutils.get_or_insert_function(builder.module, fflush_fnty, name="fflush")
+    else:
+        fflush_fn = builder.module.get_or_insert_function(fflush_fnty, name="fflush")
```
Comment (pearu) [inline rbc/externals/stdio.py:20]: Use:
```
fflush_fn = irtools.get_or_insert_function(builder.module, fflush_fnty, name='fflush')
```
(see below)

**Review by pearu — CHANGES_REQUESTED** [review]
Thanks, @guilhermeleobas for adding numba 0.54 support!

I have a couple of code quality suggestions such as introducing numba version dependent hooks into irtools.py so that other parts of the code would not need to deal with numba version checks.

File: rbc/structure_type.py [hunk 2]
```diff
@@ -130,19 +130,22 @@ def model__init__(self, dmm, fe_type):
...
 
-    ptr = builder.gep(struct.operands[0], [int32_t(0), int32_t(0)])
-    return builder.load(builder.gep(ptr, [int32_t(index)]))
+    if struct.opname == 'load':
+        ptr = builder.gep(struct.operands[0], [int32_t(0), int32_t(0)])
+        return builder.load(builder.gep(ptr, [int32_t(index)]))
+    elif struct.opname == 'insertvalue':
+        return builder.extract_value(struct, [index])
+    else:
+        raise ValueError(struct.opname)
```
[the remark this answers is not in the document]
Comment (guilhermeleobas) [reply rbc/structure_type.py:141]: `index` is a scalar value, thus, `builder.extract_value(struct, [index])` is the correct to build the instruction

File: .conda/environment.yml [hunk 3]
```diff
@@ -6,6 +6,7 @@
 
 name: rbc
 channels:
+  - numba
```
[the remark this answers is not in the document]
Comment (guilhermeleobas) [reply .conda/environment.yml:9]: Sure! I guess the `numba` channel is only need for this candidate release. After that, we can remove it.

File: rbc/structure_type.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (pearu) [reply rbc/structure_type.py:141]: Is `extract_value` path used only when using `numba>=0.54`? If so, and considering the usage of `irutils.get_member_value` elsewhere, the code here does not follow the same pattern. I wonder if this indicates a bug or did we have the code wrong for `numba<0.54`?

[the remark this answers is not in the document]
Comment (guilhermeleobas) [reply rbc/structure_type.py:141]: I guess they're equivalent. The current approach only emit a few more instructions but the tests pass with
```python
@lowering_registry.lower_getattr_generic(StructureNumbaType)
def StructureNumbaType_getattr_impl(context, builder, sig, struct, attr):
    model = datamodel.default_manager.lookup(sig)
    index = model.get_field_position(attr)
    return get_member_value(builder, struct, index)
```

**Review by pearu — APPROVED** [review]
LGTM!

I have rebased the PR with the current master and disabled some tests that require https://github.com/omnisci/omniscidb-internal/pull/5809 when using cuda-enabled omniscidb server.

The PR can be merged when CI is green.

**Pull request merged**